vignettes/standardize_data.Rmd

```diff
@@ -50,22 +52,185 @@ Standardization is an important step and extra caution is required in **repeated
+library(dplyr)
+
+# Download the 'emotion' dataset
+load(url("https://raw.github.com/neuropsychology/psycho.R/master/data/emotion.rda"))  
+  
+# Discard neutral pictures (keep only negative)
+df <- emotion %>% 
+  filter(Emotion_Condition == "Negative")  
+
+# Summary
+df %>% 
+  group_by(Participant_ID) %>% 
+  summarise(n_Trials = n(),
+            Valence_Mean = mean(Subjective_Valence, na.rm=TRUE),
+            Valence_SD = sd(Subjective_Valence, na.rm=TRUE),
+            Autobiographical_Link_Mean = mean(Autobiographical_Link, na.rm=TRUE),
+            Autobiographical_Link_SD = sd(Autobiographical_Link, na.rm=TRUE))
+```
+
+As we can see from the means and SDs, there is a lot of variability **between and within** participants (between their means and their individual SD).
 
-[Use this](https://github.com/neuropsychology/psycho.R/blob/master/docs/_posts/preparation/published/standardize_grouped_df.Rmd)
 
-<!-- **While the three standardization methods roughly present the same characteristics at a general level (mean 0 and SD 1) and a similar distribution, their values are very different and completely uncorrelated!** -->
+# Standardize
+
+We will create three dataframes standardized with each of the three techniques.
+
+```{r, fig.width=7, fig.height=4.5, eval = TRUE, results='markup', fig.align='center', comment=NA, message=FALSE, warning=FALSE}
+library(effectsize)
+
+Z_VariableWise <- df %>% 
+  standardize()
+
+Z_ParticipantWise <- df %>% 
+  group_by(Participant_ID) %>% 
+  standardize() 
+
+Z_Full <- df %>% 
+  group_by(Participant_ID) %>% 
+  standardize() %>% 
+  ungroup() %>% 
+  standardize() 
+```
+
+# Effect of Standardization 
+
+Let's see how these three standardization techniques affected the **Valence** variable.
+
+## At a general level
+```{r, fig.width=7, fig.height=4.5, eval = TRUE, results='markup', fig.align='center', comment=NA, message=FALSE, warning=FALSE}
+# Create a convenient function to print
+print_summary <- function(data){
+  paste0(
+    paste0(deparse(substitute(data)), ": "), 
+    paste(round(mean(data[["Subjective_Valence"]]), 3), 
+          "+-", 
+          round(sd(data[["Subjective_Valence"]]), 3)),
+    paste0(" [", 
+           round(min(data[["Subjective_Valence"]]), 3),
+           ",", 
+           round(max(data[["Subjective_Valence"]]), 3),
+           "]")
+    )
+}
+
+# Check the results
+print_summary(Z_VariableWise)
+print_summary(Z_ParticipantWise)
+print_summary(Z_Full)
+```
+
+The **means** and the **SD** appear as fairly similar (0 and 1)...
+
+##  Distribution
+```{r, fig.width=7, fig.height=4.5, eval = TRUE, results='markup', fig.align='center', comment=NA, message=FALSE, warning=FALSE}
+library(bayestestR)
+library(see)
+
+data.frame(VarWise = Z_VariableWise$Subjective_Valence,
+           ParWise = Z_ParticipantWise$Subjective_Valence,
+           Full = Z_Full$Subjective_Valence) %>% 
+  estimate_density(method="kernSmooth") %>% 
+  plot() +
+  see::theme_modern()
```

Comment:
@DominiqueMakowski can't get this bit to work.
I have dev version of see, but see no `see:::plot.see_estimate_density_df` method there.

Reply:
try with the master `see`? Although having a nice `NEWS`, `see` has sometimes a dev behind master 🤷‍♂

Reply:
I think it's related to a recent change that we made with @strengejacke

Reply:
``` r
library(easystats)
#> # Attaching packages
#> <U+2714> insight     0.8.2.1   <U+2714> bayestestR  0.5.3  
#> <U+2714> performance 0.4.4.1   <U+2714> parameters  0.6.0.1
#> <U+2714> see         0.4.1.1   <U+2714> effectsize  0.3.0  
#> <U+2714> correlation 0.1.1     <U+2714> modelbased  0.1.2  
#> <U+2714> report      0.1.0
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

iris %>% 
  estimate_density(method = "kernSmooth") %>% 
  plot() +
  theme_modern()
#> Loading required namespace: KernSmooth
```

![](https://i.imgur.com/Pd2QHWM.png)

``` r

iris %>% 
  group_by(Species) %>% 
  estimate_density(method = "kernSmooth") %>% 
  plot() +
  theme_modern()
```

![](https://i.imgur.com/r2bhVqc.png)

<sup>Created on 2020-03-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>